packages/react-day-picker/src/contexts/Focus/FocusContext.tsx

```diff
@@ -59,35 +96,148 @@ export function FocusProvider({
   const focusDayBefore = () => {
     if (!focusedDay) return;
     const before = addDays(focusedDay, -1);
-    setDay(before);
+    changeInitialFocus(before);
     switchMonth(before, numberOfMonths * -1);
   };
   const focusDayAfter = () => {
     if (!focusedDay) return;
     const after = addDays(focusedDay, 1);
-    setDay(after);
+    changeInitialFocus(after);
     switchMonth(after, numberOfMonths);
   };
   const focusWeekBeforeDay = () => {
     if (!focusedDay) return;
     const up = addWeeks(focusedDay, -1);
-    setDay(up);
+    changeInitialFocus(up);
     switchMonth(up, numberOfMonths * -1);
   };
   const focusWeekAfterDay = () => {
     if (!focusedDay) return;
     const down = addWeeks(focusedDay, 1);
-    setDay(down);
+    changeInitialFocus(down);
     switchMonth(down, numberOfMonths);
   };
 
+  const focusFirstDayOfWeek = (): void => {
+    if (!focusedDay) return;
+    const dayToFocus = addDays(
+      startOfWeek(addDays(focusedDay, -1), {
+        weekStartsOn: 0
+      }),
+      1
+    );
```

Comment:
Why going to the day before? (maybe is right)

```suggestion
  const focusFirstDayOfWeek = (): void => {
    if (!focusedDay) return;
    const dayToFocus = startOfWeek(focusedDay);
```

Reply:
For some reason, date fns lib makes it so going to the start/end of the week returns one day after the beginning/end of the week so that's why I have one additional day added

Reply:
> going to the start/end of the week returns one day after the beginning/end of the week

Do they go at 0.00 and 23:59 respectively? Worth investigating.

This is sure worth investigating in another instance